examples/basic/interface/src/ianimal.rs

```diff
@@ -1,6 +1,8 @@
 use com::{com_interface, interfaces::iunknown::IUnknown, sys::HRESULT};
 
-#[com_interface("EFF8970E-C50F-45E0-9284-291CE5A6F771")]
-pub trait IAnimal: IUnknown {
-    unsafe fn eat(&self) -> HRESULT;
+com_interface! {
+    #[uuid("EFF8970E-C50F-45E0-9284-291CE5A6F771")]
+    pub unsafe interface IAnimal: IUnknown {
+        unsafe fn eat(&self) -> HRESULT;
```

Comment:
I think COM interfaces should just be implicitly `pub unsafe`. When it comes to the more interesting examples, it quickly becomes very tedious to repeat this. I should just be able to write:

```rust
#[uuid("EFF8970E-C50F-45E0-9284-291CE5A6F771")]
interface IAnimal: IUnknown {
    fn eat(&self) -> HRESULT;
}
```

Reply:
I’m not so sure. `unsafe` is always just boilerplate, it has no effect on the runtime characteristics of a program. That’s the entire point. It’s there to make the dev stop and think, and to give a place to easily audit. 

Now I could be convinced to leave off unsafe for method declarations since they are unsafe to call, but I think having it on the Interface itself would be good. You’re then declaring that there is a vtable that matches your declaration for the given UUID.

Reply:
I guess it depends on who is writing the declaration. I agree having it there is useful, but writing it for dozens of interfaces with dozens of methods is tedious. But then writing these by hand should be rare.

Reply:
Having written many of these by hand myself, adding unsafe was probably the least tedious aspect 😅 I think the benefit of unsafe auditing is worth the small pain. I am still on the fence about the method declarations.

Reply:
Isn't that implied by the interface as a whole being unsafe?

Reply:
The semantics of unsafe traits are “there are invariants you must check for implementing this trait safely” (an example is Sync which requires you check that the type can be shared across threads). Unsafe methods require you check invariants for any time you call the the method. Unsafe methods can exist of safe traits and unsafe traits can have safe methods

Reply:
How this maps to interfaces is up to us since that’s not a rust concept, but declaring an interface involves automatically implementing unsafe traits so I think marking the interface unsafe makes sense